camayoc/tests/qpc/cli/csv_report_parsing.py

```diff
@@ -0,0 +1,135 @@
+# coding=utf-8
+"""Helper functions for parsing csv reports."""
+
+import csv
+
+EXPECTED_SUMMARY_REPORT_ID_FIELDS = [
+    'Report ID',
+    'Report Type',
+    'Report Version',
+    'Report Platform ID',
+]
+
+EXPECTED_DETAIL_REPORT_ID_FIELDS = [
+    'Report ID',
+    'Report Type',
+    'Report Version',
+    'Report Platform ID',
+    'Number Sources'
+]
+
+
+def normalize_csv_report(f, header_range, header_lines, report_type='summary'):
+    """Extract and normalize csv report to match the returned JSON report.
+
+    :param f: A file object for the csv
+    :param header_range: An int specifing the range that the head extends into
+        the csv file.
+    :param header_lines: A list of tuples containg the index pairs for the
+        lines to be used as the key/value pairs of each header info dictionary
+        (ex: [(0,1), (3,4)] ).
+    :param report_type: A string that defines what type of report object to
+        return, 'summary' or 'detail'. Defaults to 'summary'.
+    """
+    # First grab the header information from the file object.
+    report_headers = [f.readline() for _ in range(header_range)]
+    # Next, extract header information into a list of dictonaries (1 dictionary
+    # per header set)
+    header_info = extract_key_value_lines(report_headers,
+                                          header_lines)
+    # Now that we extracted the report information we can use CSV reader to
+    # read the system fingerprints information
+    reader = csv.DictReader(f)
+    if report_type == 'summary':
+        report = normalize_summary_report(header_info, reader)
+    else:
+        report = normalize_detail_report(header_info, reader)
+    return report
```

Comment:
If you want you can drop this line and do something like this:


```python
    if report_type == 'summary':
        return normalize_summary_report(header_info, reader)
    return normalize_detail_report(header_info, reader)
```

Or if you prefer having the full if/else block:

```python
    if report_type == 'summary':
        return normalize_summary_report(header_info, reader)
    else:
        return normalize_detail_report(header_info, reader)
```